Kernel/Devices/HID/PS2/MouseDevice.cpp

```diff
@@ -156,11 +156,25 @@ UNMAP_AFTER_INIT ErrorOr<NonnullOwnPtr<PS2MouseDevice>> PS2MouseDevice::try_to_i
     return device;
 }
 
-UNMAP_AFTER_INIT ErrorOr<void> PS2MouseDevice::initialize()
+UNMAP_AFTER_INIT ErrorOr<int> PS2MouseDevice::reset_device()
 {
-    TRY(attached_controller().reset_device(attached_port_index()));
+    auto do_reset = [this] -> ErrorOr<int> {
+        TRY(attached_controller().reset_device(attached_port_index()));
+        return TRY(read_from_device());
+    };
+    auto maybe_device_id = do_reset();
+    for (int attempt = 1; attempt < 10; attempt++) {
+        if (!maybe_device_id.is_error())
+            return maybe_device_id;
+        microseconds_delay(500);
```

Comment:
That's a lot of microseconds. For 10 retries, that's up to 5s. Do shorter delays help less? Do we really need that much time?

Reply:
Mixed up microseconds and milliseconds? 😅 5000 microseconds = 5 milliseconds

Reply:
😬